lib/Twig/Extensions/Node/Trans.php

```diff
@@ -41,6 +41,10 @@ public function compile(Twig_Compiler $compiler)
 
         $function = null === $this->getNode('plural') ? 'gettext' : 'ngettext';
 
+        if (null !== $this->getNode('domain')) {
```

Comment:
@sagikazarmark As soon as you use the ``{% plural %}`` tag in your template, this is forced to be a pluralized message. So the presence of the node is enough to switch from one code path to the other.

As you allow any expression for the domain, it would make sense to account for ``null`` there though, using the default domain